.github/workflows/simulation_test.yml

```diff
@@ -17,15 +17,32 @@ on:
 
 jobs:
   simulation:
-    runs-on: ubuntu-22.04
+    runs-on: ${{ matrix.os }}
 
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        python-version: [3.9, 3.12]
     steps:
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
     - name: Checkout neurodamus repository
       uses: actions/checkout@v4
       with:
         ref: ${{ github.event.pull_request.head.sha }}
         fetch-depth: 0
 
+    - name: Cache pip
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ matrix.python-version }}-v1
```

Comment:
Here `${{ runner.os }}` = `Linux` , but I feel it is not sufficient for the key. How about using ${{ matrix.os }} which is `ubuntu-22.04` ?

Reply:
sounds good!